### PR TITLE
Update the versioning logic for the Godot Android Editor

### PR DESCRIPTION
**Pull request by m4gr3d**
This is necessary to separate subsequent uploads to the Google Play store as each upload needs to increment the version code.

Also updates the set of supported screen size to exclude small devices (e.g: watches). Affects less than **0.5%** of supported devices.

Screen Shot 2022-09-12 at 7 29 55 AM

[3.x version](https://github.com/godotengine/godot/pull/65683)

### CODE REVIEW

File: platform/android/java/editor/build.gradle [hunk 1]
```diff
@@ -12,6 +12,25 @@ dependencies {
...
+    // the Google Play store.
+    // This should be reset on each stable release of Godot.
+    editorBuildNumber = 0
+    // Value by which the Godot version code should be offset by to make room for the build number
+    editorBuildNumberOffset = 1000
+}
+
+def generateVersionCode() {
+    int libraryVersionCode = getGodotLibraryVersionCode()
+    return (libraryVersionCode * editorBuildNumberOffset) + editorBuildNumber
```
Comment (akien-mga) [inline platform/android/java/editor/build.gradle:26]: To be sure, this is used as a signed 64-bit int? Because we're getting close to `INT32_MAX` already with the numbers chosen (`(4 * 100000) * 1000` is 400,000,000, `INT32_MAX` is `2,100,000,000` and peanuts).

**Comment by akien-mga** [comment]
> This is necessary to separate subsequent uploads to the Google Play store as each upload needs to increment the version code.

I don't think this change will be sufficient then. We usually have multiple alpha / beta / RC builds, but as I understand it this change only gives you the possibility to upload a single build for each. So if you upload 3.6 beta 1, then you can't do updates until the 3.x branch is ready to move to RC.

File: platform/android/java/editor/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -7,7 +7,7 @@
     <supports-screens
         android:largeScreens="true"
         android:normalScreens="true"
-        android:smallScreens="true"
+        android:smallScreens="false"
```
[the remark this answers is not in the document]
Comment (m4gr3d) [reply platform/android/java/editor/src/main/AndroidManifest.xml:10]: > Sneaky unrelated change ;)

Yeah, I should have made it a separate commit :). I'd rephrase this PR as addressing some of the issues I encountered while setting up the play store release.

> Does that mean we won't be able to install the Android editor on phones?

No, `small` is for devices even smaller than regular phones (most regular phones fall under the `normal` or larger category). Here's a reference screenshot of what some of those devices are. From the play store info, this affects less than **0.5%** of supported devices.

Screen Shot 2022-09-12 at 7 29 55 AM

File: platform/android/java/editor/build.gradle (hunk 1 above)
Comment (m4gr3d) [reply platform/android/java/editor/build.gradle:26]: You're right, according to the [doc](https://developer.android.com/studio/publish/versioning#appversioning):
> The greatest value Google Play allows for versionCode is 2100000000.

I'll reduce the range for the build number.

**Comment by m4gr3d** [comment]
> Can we pass this as an argument to scons or gradle at build time

Yeah, I'll create an issue to track the task. 

We can't automate the upload process until the app is first reviewed and released to the store, so at the moment the process will be manual for which the current approach should work.

Once we're able to fully automated the upload process, then we should update it to pass the build number at build time.

**Comment by m4gr3d** [comment]
> Yeah, I'll create an issue to track the task.

Added https://github.com/godotengine/godot-proposals/issues/5410 to track the issue.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!
